### PR TITLE
Adding arm64 support

### LINKED ISSUE #59: M1 mac arm64 build
Would it be possible to provide builds for the M1 mac?

```sh
✗ kubectl krew install neat
Updated the local copy of plugin index.
Installing plugin: neat
W0321 13:58:17.153568   38001 install.go:164] failed to install plugin "neat": plugin "neat" does not offer installation for this platform
F0321 13:58:17.153611   38001 root.go:79] failed to install some plugins: [neat]: plugin "neat" does not offer installation for this platform
```

Thanks!

### PR DESCRIPTION
**Pull request by jmcshane**
Resolves #59

### CODE REVIEW

**Comment by jmcshane** [comment]
ran this build locally with the `--snapshot` flag successfully.

need to run the build with go 1.16 to provide that support, so we may need to update `.github/workflows/ci.yml`

**Comment by jmcshane** [comment]
hmm, ok i see we've got some issues with the `Makefile`. ill get this PR up to date

**Comment by jmcshane** [comment]
ok, the make targets and e2e tests are working on my M1 machine.

**Comment by jmcshane** [comment]
i don't have a kubernetes cluster on this machine yet (`kind` isn't working on here yet), so you may need to validate `make test-integration` from this PR

**Comment by itaysk** [comment]
@jmcshane This is great! thank you

File: Makefile [hunk 1]
```diff
@@ -8,24 +8,25 @@
...
 	bats ./test/e2e-kubectl.bats
 	bats ./test/e2e-krew.bats
 
-build: dist/kubectl-neat_$(os)
+build: dist/kubectl-neat_$(os)_$(arch)
 
 SRC = $(shell find . -type f -name '*.go' -not -path "./vendor/*")
 dist/kubectl-neat_%: $(SRC)
-	GOOS=$* go build -o dist/$(@F)
+	GOOS=$(os) GOARCH=$(arch) go build -o dist/$(@F)
```
Comment (itaysk) [inline Makefile:29]: In order to support cross-compilation, this target should take into account the *requested* os/arch, while it currently uses the *runtime* os/arch.

Comment (jmcshane) [reply Makefile]: @itaysk I added a function to split the `$*` argument into _requested_ os/arch for the build flags

**Review by itaysk — APPROVED** [review]

**Pull request merged**